src/msg/simple/Pipe.cc

```diff
@@ -466,6 +466,16 @@ int Pipe::accept()
     existing = msgr->_lookup_pipe(peer_addr);
     if (existing) {
       existing->pipe_lock.Lock(true);  // skip lockdep check (we are locking a second Pipe here)
+      if (existing->state == STATE_CLOSED) {
+        ldout(msgr->cct, 1) << __func__ << " existing racing replace or mark_down happened while replacing."
+                            << " existing_state=" << existing->get_state_name() << dendl;
+	reply.tag = CEPH_MSGR_TAG_RETRY_GLOBAL;
```

Comment:
why not send a RETRY_SESSION message and ask the peer to bump up its cseq? so its cseq won't collide with the existing one. if this makes sense to you, maybe we can reuse the "retry_session:", in other words
```diff
--- a/src/msg/simple/Pipe.cc
+++ b/src/msg/simple/Pipe.cc
@@ -524,11 +524,12 @@ int Pipe::accept()
 
       if (connect.connect_seq == existing->connect_seq) {
 	// if the existing connection successfully opened, and/or
-	// subsequently went to standby, then the peer should bump
+	// subsequently went to standby/closed, then the peer should bump
 	// their connect_seq and retry: this is not a connection race
 	// we need to resolve here.
 	if (existing->state == STATE_OPEN ||
-	    existing->state == STATE_STANDBY) {
+	    existing->state == STATE_STANDBY ||
+	    existing->state == STATE_CLOSED) {
 	  ldout(msgr->cct,10) << "accept connection race, existing " << existing
 			      << ".cseq " << existing->connect_seq
 			      << " == " << connect.connect_seq
```